Lib/dataclasses.py

```diff
@@ -1101,6 +1104,14 @@ def _process_class(cls, init, repr, eq, order, unsafe_hash, frozen,
     if slots:
         cls = _add_slots(cls, frozen)
 
+    # Frozen __setattr__/__delattr__ must be added after __slots__
+    # bpo-45897
+    if frozen:
+        for fn in _frozen_get_del_attr(cls, field_list, globals):
```

Comment:
I think this function ought to be named `_frozen_set_del_attr`. That's outside the scope of this PR though.